mgmtd/mgmt_be_adapter.c

```diff
@@ -34,55 +34,30 @@
  */
 struct mgmt_be_xpath_map {
 	char *xpath_regexp;
-	uint subscr_info[MGMTD_BE_CLIENT_ID_MAX];
-};
-
-struct mgmt_be_client_xpath {
-	const char *xpath;
-	uint subscribed;
-};
-
-struct mgmt_be_client_xpath_map {
-	struct mgmt_be_client_xpath *xpaths;
-	uint nxpaths;
-};
-
-struct mgmt_be_get_adapter_config_params {
-	struct mgmt_be_client_adapter *adapter;
-	struct nb_config_cbs *cfg_chgs;
-	uint32_t seq;
+	uint64_t clients;
 };
 
 /*
  * Each client gets their own map, but also union all the strings into the
  * above map as well.
  */
 #if HAVE_STATICD
-static struct mgmt_be_client_xpath staticd_xpaths[] = {
-	{
-		.xpath = "/frr-vrf:lib/*",
-		.subscribed = MGMT_SUBSCR_VALIDATE_CFG | MGMT_SUBSCR_NOTIFY_CFG,
```

Comment:
Removing this kind of granularity will not suffice in scenarios like following:
- A XPATH in one of the config datastore is NOT to be sent for validation/notifciation to the specific client, but the same needs to be retrieved through the operational datastore. For example lib-interface/* and lib-vhf/* are only to be retrieved as operational data from Clients like static or zebra. But the config is to be only passed to zebra. Rest of the other daemons get the interface and vrf information through lib/interface.c and lib/vrf.c libraries from zebra. You can refer to 
We need at least this much granularity in the subscription map.

Reply:
Ther operational datastore uses a different set of xpaths

Reply:
static const char **be_client_oper_xpaths[] = {};

Reply:
There's no reason to walk though all the xpaths for config looking for interleaved operation state xpaths. They are never needed together, we are iether dealing with op state or we are dealing with config state. So I have them in 2 different arrays.

Reply:
The above statement is with the assumption that for the same item the config xpaths and operstate paths are different. However even though that is true for certain models e.g. Openconfig) we can't say the same about the current FRR models. To achieve the same a major rehaul would be necessary in quite a few of the Yang models in FRR. I haven't seen any example yet where the same item has separate config and overstate paths in FRR yang models.

Reply:
The above statement makes no such assumption.

Reply:
Perhaps you are not understanding my comment. I am saying that when mgmtd needs to get to op-state it has no interest in the config xpaths, likewise when mgmtd needs to match config state xpaths it has no interest in op-state matching. That is all, this has nothing to do with yang model design.

Reply:
I mean look at the code your commenting on, and you can see this.

Reply:
Ok I missed that part in the diffs that you are introducing two different maps now.

Even then for Config we will probably need to differentiate between the client that will need to validate a configuration item from clients who just needs to be notified about the same (after the validator has validated). Today with limited clients we have support on, this scenario has not come up yet. But we may come across this in future. Now I know you this will be over-engineering for you and lot of other members. So I will not make a fuss now.

But my take on this PR is still that this simplification is really not that necessary. Keeping the map this way leaves a lot of scope for future enhancements. And we are really not gaining much with this simplification.

Reply:
I believe the change request you made is resolved now.